Subject: [PATCH] Wifi: Restore hidden flag for hidden networks on update network

Currently when user modifies any of the network parameter for hidden
networks, hidden flag is not updated. As a result this network is no
longer part of hidden networks, we are unable to reconnect when user
triggers wifi off and on.

This commit sets the value of hiddenSSID flag of config at the Settings
side on network update.

diff --git a/src/com/android/settings/wifi/WifiConfigController.java b/src/com/android/settings/wifi/WifiConfigController.java
index ecd4b5966e..d7bccc0d07 100644
--- a/src/com/android/settings/wifi/WifiConfigController.java
+++ b/src/com/android/settings/wifi/WifiConfigController.java
@@ -517,6 +517,7 @@ void showWarningMessagesIfAppropriate() {
                     mAccessPoint.getSsidStr());
         } else {
             config.networkId = mAccessPoint.getConfig().networkId;
+            config.hiddenSSID = mAccessPoint.getConfig().hiddenSSID;
         }
 
         config.shared = mSharedCheckBox.isChecked();
